include/llvm/IR/BasicBlock.h

```diff
@@ -245,6 +245,8 @@ class BasicBlock : public Value, // Basic blocks are data objects also
   inline const Instruction       &back() const { return InstList.back();  }
   inline       Instruction       &back()       { return InstList.back();  }
 
+  size_t size_no_dbg() const; // HLSL Change - Get the size of the block without the debug insts
```

Comment:
I suppose my only issue is that it's not clear that this is an expensive operation compared to size().  Perhaps it should be something like compute_size_no_dbg() to indicate that it's not just a cheap getter, and shouldn't be used frequently.